extensions/blocks/calendly/calendly.php

```diff
@@ -114,13 +115,26 @@ function load_assets( $attr, $content ) {
 				esc_attr( $classes ),
 				esc_attr( $block_id )
 			);
-			$script  = <<<JS_END
-Calendly.initInlineWidget({
-	url: '%s',
-	parentElement: document.getElementById('%s'),
-	inlineStyles: false,
-});
+
+			$init_calendly_widget_js = "Calendly.initInlineWidget({
+				url: '%s',
+				parentElement: document.getElementById('%s'),
+				inlineStyles: false,
+			});";
+
+			if ( $is_p2_site ) {
+				// For P2s only: wait until after o2 has
+				// replaced main#content to initialize widget.
+				$script = <<<JS_END
+jQuery('body').on( 'ready.o2', function( event, domRef ) {
```

Comment:
`jetpack-calendly-external-js` doesn't rely on jQuery, so it may be best to try to use vanilla JS here if possible.

Reply:
> `jetpack-calendly-external-js` doesn't rely on jQuery, so it may be best to try to use vanilla JS here if possible.

Unfortunately, o2 events are not visible to native DOM handlers. ☹️ jQuery is a dependency of o2, and will always be available if o2 is active.

Reply:
I think now it's properly handled, considering that

* `o2` depends totally on jQuery
* it checks whether both window propertties are defined:

```js
if( window.jQuery && window.o2 ) {
  // ...
}
```